Fetch only lines when the app starts

This will speed up initial home page load.

Try to fix the error with cypress tests (happens when fetchAllLines query gets canceled):
VM11015 bundle.js:123317 Error: Network error: Failed to fetch
    at new ApolloError (VM9162 bundle.js:7440)
    at VM9162 bundle.js:8624
    at VM9162 bundle.js:8936
    at Set.forEach (<anonymous>)
    at VM9162 bundle.js:8934
    at Map.forEach (<anonymous>)
    at QueryManager../node_modules/apollo-client/bundle.esm.js.QueryManager.broadcastQueries (VM9162 bundle.js:8932)
    at VM9162 bundle.js:8538

--- a/cypress/integration/linkView.spec.js
+++ b/cypress/integration/linkView.spec.js
@@ -25,6 +25,8 @@ describe('LinkView tests - write access user', () => {
         cy.getTestElement('editButton').should('exist');
         cy.getTestElement('editButton').click();
 
+        cy.getTestElement('speed').clear().type(50);
+
         cy.getTestElement('measuredLength')
             .invoke('val')
             .then((value) => {

--- a/src/components/App.tsx
+++ b/src/components/App.tsx
@@ -6,14 +6,18 @@ import { matchPath, Router, Switch } from 'react-router';
 import { Route } from 'react-router-dom';
 import EndpointPath from '~/enums/endpointPath';
 import LocalStorageHelper from '~/helpers/LocalStorageHelper';
+import { ISearchLine } from '~/models/ILine';
 import navigator from '~/routing/navigator';
 import SubSites from '~/routing/subSites';
 import AuthService, { IAuthorizationResponse } from '~/services/authService';
 import CodeListService from '~/services/codeListService';
+import LineService from '~/services/lineService';
 import { CodeListStore } from '~/stores/codeListStore';
 import { ErrorStore } from '~/stores/errorStore';
 import { LoginStore } from '~/stores/loginStore';
 import { MapStore } from '~/stores/mapStore';
+import { SearchResultStore } from '~/stores/searchResultStore';
+import { SearchStore } from '~/stores/searchStore';
 import HttpUtils from '~/utils/HttpUtils';
 import '~/utils/KeyEventHandler';
 import AppFrame from './AppFrame';
@@ -28,6 +32,8 @@ interface IAppState {
 interface IAppProps {
     loginStore?: LoginStore;
     mapStore?: MapStore;
+    searchStore?: SearchStore;
+    searchResultStore?: SearchResultStore;
     codeListStore?: CodeListStore;
     errorStore?: ErrorStore;
 }
@@ -35,7 +41,7 @@ interface IAppProps {
 const browserHistory = createBrowserHistory();
 const history = syncHistoryWithStore(browserHistory, navigator.getStore());
 
-@inject('mapStore', 'loginStore', 'codeListStore', 'errorStore')
+@inject('mapStore', 'loginStore', 'codeListStore', 'searchStore', 'searchResultStore', 'errorStore')
 @observer
 class App extends React.Component<IAppProps, IAppState> {
     constructor(props: IAppProps) {
@@ -76,11 +82,23 @@ class App extends React.Component<IAppProps, IAppState> {
         this.setState({
             isAppInitializationInProgress: true,
         });
-        await this.initCodeLists();
-        await this.fetchSaveLock();
-        this.setState({
-            isAppInitializationInProgress: false,
+        Promise.all([this.fetchAllLines(), this.initCodeLists(), this.fetchSaveLock()]).then(() => {
+            this.setState({
+                isAppInitializationInProgress: false,
+            });
         });
+    };
+
+    private fetchAllLines = async () => {
+        this.props.searchStore!.setIsLoading(true);
+        try {
+            const searchLines: ISearchLine[] = await LineService.fetchAllSearchLines();
+            this.props.searchResultStore!.setAllLines(searchLines);
+            this.props.searchResultStore!.search();
+        } catch (e) {
+            this.props.errorStore!.addError('Linjojen haku ei onnistunut', e);
+        }
+        this.props.searchStore!.setIsLoading(false);
     };
 
     private initCodeLists = async () => {

--- a/src/components/shared/searchView/SearchResults.tsx
+++ b/src/components/shared/searchView/SearchResults.tsx
@@ -6,7 +6,6 @@ import { ISearchLine } from '~/models/ILine';
 import { ISearchNode } from '~/models/INode';
 import Navigator from '~/routing/navigator';
 import subSites from '~/routing/subSites';
-import LineService from '~/services/lineService';
 import NodeService from '~/services/nodeService';
 import { ErrorStore } from '~/stores/errorStore';
 import { SearchResultStore } from '~/stores/searchResultStore';
@@ -23,7 +22,6 @@ interface ISearchResultsProps {
 }
 
 interface ISearchResultsState {
-    isLoading: boolean;
     showLimit: number;
 }
 
@@ -41,7 +39,6 @@ class SearchResults extends React.Component<ISearchResultsProps, ISearchResultsS
     constructor(props: ISearchResultsProps) {
         super(props);
         this.state = {
-            isLoading: true,
             showLimit: SHOW_LIMIT_DEFAULT,
         };
 
@@ -50,7 +47,6 @@ class SearchResults extends React.Component<ISearchResultsProps, ISearchResultsS
 
     async componentDidMount() {
         this.showMoreResults();
-        this.fetchAll();
         this.reactionDisposer = reaction(
             () => [
                 this.props.searchStore!.searchInput,
@@ -58,33 +54,20 @@ class SearchResults extends React.Component<ISearchResultsProps, ISearchResultsS
             ],
             this.scrollToBeginning
         );
+        // Lazy load nodes in the background
+        this.fetchAllNodes();
     }
 
     componentWillUnmount() {
         this.reactionDisposer();
     }
 
-    private fetchAll = async () => {
-        this.setState({ isLoading: true });
-        return Promise.all([this.fetchAllLines(), this.fetchAllNodes()]).then(() => {
-            this.setState({ isLoading: false });
-            this.props.searchResultStore!.search();
-        });
-    };
-
-    private fetchAllLines = async () => {
-        try {
-            const searchLines: ISearchLine[] = await LineService.fetchAllSearchLines();
-            this.props.searchResultStore!.setAllLines(searchLines);
-        } catch (e) {
-            this.props.errorStore!.addError('Linjojen haku ei onnistunut', e);
-        }
-    };
-
     private fetchAllNodes = async () => {
         try {
-            const nodes = await NodeService.fetchAllNodes();
-            this.props.searchResultStore!.setAllNodes(nodes);
+            NodeService.fetchAllNodes().then((nodes: any) => {
+                this.props.searchResultStore!.setAllNodes(nodes);
+                this.props.searchResultStore!.search();
+            });
         } catch (e) {
             this.props.errorStore!.addError('Solmujen haku ei onnistunut', e);
         }
@@ -130,7 +113,7 @@ class SearchResults extends React.Component<ISearchResultsProps, ISearchResultsS
         const filteredNodes = this.getFilteredNodes();
         return (
             <div className={s.searchResultsView}>
-                {this.state.isLoading ? (
+                {searchStore!.isLoading ? (
                     <div className={s.searchResultsView}>
                         <Loader />
                     </div>

--- a/src/components/shared/searchView/SearchResults.tsx
+++ b/src/components/shared/searchView/SearchResults.tsx
@@ -63,8 +63,10 @@ class SearchResults extends React.Component<ISearchResultsProps, ISearchResultsS
     }
 
     private fetchAllNodes = async () => {
+        if (this.props.searchResultStore!.allNodes.length > 0) return;
+
         try {
-            NodeService.fetchAllNodes().then((nodes: any) => {
+            await NodeService.fetchAllNodes().then((nodes: any) => {
                 this.props.searchResultStore!.setAllNodes(nodes);
                 this.props.searchResultStore!.search();
             });

--- a/src/components/sidebar/linkView/LinkView.tsx
+++ b/src/components/sidebar/linkView/LinkView.tsx
@@ -330,6 +330,7 @@ class LinkView extends React.Component<ILinkViewProps, ILinkViewState> {
                                 type='number'
                                 validationResult={invalidPropertiesMap['speed']}
                                 onChange={this.onChangeLinkProperty('speed')}
+                                data-cy='speed'
                             />
                         </div>
                         {!this.props.isNewLink && (

--- a/src/services/graphqlQueries.ts
+++ b/src/services/graphqlQueries.ts
@@ -255,6 +255,7 @@ const getAllNodesQuery = () => {
                     solkirjain
                     transitTypes
                     pysakkiBySoltunnus {
+                        soltunnus
                         pysnimi
                     }
                 }

--- a/src/stores/searchStore.ts
+++ b/src/stores/searchStore.ts
@@ -8,6 +8,7 @@ class SearchStore {
     @observable private _isSearchingForNodes: boolean;
     @observable private _areInactiveLinesHidden: boolean;
     @observable private _isSearchDisabled: boolean;
+    @observable private _isLoading: boolean;
 
     constructor() {
         this._searchInput = '';
@@ -22,6 +23,7 @@ class SearchStore {
         this._isSearchingForNodes = false;
         this._areInactiveLinesHidden = true;
         this._isSearchDisabled = false;
+        this._isLoading = false;
     }
 
     @computed
@@ -59,6 +61,11 @@ class SearchStore {
         return this._isSearchDisabled;
     }
 
+    @computed
+    get isLoading() {
+        return this._isLoading;
+    }
+
     @action
     public toggleIsSearchingForLines() {
         this._isSearchingForLines = true;
@@ -89,6 +96,11 @@ class SearchStore {
     @action
     public setIsSearchDisabled(isSearchDisabled: boolean) {
         this._isSearchDisabled = isSearchDisabled;
+    }
+
+    @action
+    public setIsLoading(isLoading: boolean) {
+        this._isLoading = isLoading;
     }
 }
 

